player/src/main/java/com/focusit/jsflight/player/script/PlayerScriptProcessor.java

```diff
@@ -154,19 +154,8 @@ public JSONObject runStepTemplating(UserScenario scenario, JSONObject step)
                 {
                     String source = result.getString(key);
 
-                    String pattern = "[^\\\\](\\$)(.)";
-                    Pattern r = Pattern.compile(pattern);
-                    Matcher m = r.matcher(source);
-                    while (m.find())
-                    {
-                        if (!m.group(1).equalsIgnoreCase("{"))
-                        {
-                            int start = m.start(0);
-                            source = source.substring(0, start + 1) + "\\$"
-                                    + source.substring(start + 2, source.length());
-                        }
-                        m = r.matcher(source);
-                    }
+                    source = StringUtils.replace(source, "$", "\\$");
```

Comment:
Why you used Common's StringUtils? Is java.lang.String#replaceAll(String regex, String replacement) not enough ?